Mathlib/Algebra/MvPolynomial/Basic.lean

```diff
@@ -238,6 +239,16 @@ instance infinite_of_nonempty (σ : Type*) (R : Type*) [Nonempty σ] [CommSemiri
   Infinite.of_injective ((fun s : σ →₀ ℕ => monomial s 1) ∘ Finsupp.single (Classical.arbitrary σ))
     <| (monomial_left_injective one_ne_zero).comp (Finsupp.single_injective _)
 
+instance noZeroDivisors [CommSemiring R] [NoZeroDivisors R] : NoZeroDivisors (MvPolynomial σ R) :=
```

Comment:
Arguably these should have `inst`; I suspect automatic name generation just works here?

(same throughout the PR)